cscs-checks/apps/lammps/lammps_check.py

```diff
@@ -7,102 +7,169 @@
-        ])
-        self.strict_check = False
-        self.extra_resources = {
-            'switches': {
-                'num_switches': 1
-            }
-        }
+    @run_after('init')
+    def set_tags(self):
+        self.tags |= {'maintenance' if self.benchmark == 'maint'
+                      else 'production'}
+
+
+@rfm.simple_test
+class LAMMPSGPUCheck(LAMMPSCheck):
+    scale = parameter(['small', 'large'])
+    benchmark = parameter(['prod', 'maint'])
+    valid_systems = ['daint:gpu']
+    executable = 'lmp_mpi'
+    input_file = 'in.lj.gpu'
+    executable_opts = ['-sf gpu', '-pk gpu 1', '-in', input_file]
+    variables = {'CRAY_CUDA_MPS': '1'}
+    num_gpus_per_node = 1
 
-        self.tags = {'scs', 'external-resources'}
-        self.maintainers = ['TR', 'VH']
-
-
-@rfm.parameterized_test(*([s, v]
-                          for s in ['small', 'large']
-                          for v in ['prod', 'maint']))
-class LAMMPSGPUCheck(LAMMPSBaseCheck):
-    def __init__(self, scale, variant):
-        super().__init__()
-        self.valid_systems = ['daint:gpu']
-        self.executable = 'lmp_mpi'
-        self.executable_opts = ['-sf gpu', '-pk gpu 1', '-in in.lj.gpu']
-        self.variables = {'CRAY_CUDA_MPS': '1'}
-        self.num_gpus_per_node = 1
-        if scale == 'small':
+    @run_after('init')
+    def set_reference(self):
+        if self.scale == 'small':
+            self.reference = REFERENCE_GPU_PERFORMANCE_SMALL
+        else:
+            self.reference = REFERENCE_GPU_PERFORMANCE_LARGE
+
+    @run_after('init')
+    def set_num_tasks(self):
+        if self.scale == 'small':
             self.valid_systems += ['dom:gpu']
             self.num_tasks = 12
             self.num_tasks_per_node = 2
         else:
             self.num_tasks = 32
             self.num_tasks_per_node = 2
 
-        references = {
-            'maint': {
-                'small': {
-                    'dom:gpu': {'perf': (3457, -0.10, None, 'timesteps/s')},
-                    'daint:gpu': {'perf': (2524, -0.10, None, 'timesteps/s')}
-                },
-                'large': {
-                    'daint:gpu': {'perf': (3832, -0.05, None, 'timesteps/s')}
-                }
-            },
-            'prod': {
-                'small': {
-                    'dom:gpu': {'perf': (3132, -0.05, None, 'timesteps/s')},
-                    'daint:gpu': {'perf': (2400, -0.40, None, 'timesteps/s')}
-                },
-                'large': {
-                    'daint:gpu': {'perf': (3260, -0.50, None, 'timesteps/s')}
-                }
-            },
-        }
-        self.reference = references[variant][scale]
-        self.tags |= {'maintenance' if variant == 'maint' else 'production'}
 
+@rfm.simple_test
+class LAMMPSCPUCheck(LAMMPSCheck):
+    scale = parameter(['small', 'large'])
+    benchmark = parameter(['prod'])
+    valid_systems = ['daint:mc', 'eiger:mc', 'pilatus:mc']
+    input_file = 'in.lj.cpu'
```

Comment:
Do you see? The input file is being defined again here. The previous GPU class also defines this.